resources/metadata/githubbranchprotection.yaml

```diff
@@ -0,0 +1,90 @@
+metadata:
+  name: githubCheckBranchProtection
+  description: Check branch protection of a GitHub branch
+  longDescription: |
+    This step allows you to check if certain branch protection rules are fulfilled.
+
+    It can for example be used to verify if certain status checks are mandatory. This can be helpful to decide if a certain check needs to be performed again after merging a pull request.
+spec:
+  inputs:
+    secrets:
+      - name: githubTokenCredentialsId
+        description: Jenkins 'Secret text' credentials ID containing token to authenticate to GitHub.
+        type: jenkins
+    params:
+      - name: apiUrl
+        aliases:
+          - name: githubApiUrl
+        description: Set the GitHub API url.
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        default: https://api.github.com
+        mandatory: true
+      - name: branch
+        description: The name of the branch for which the protection settings should be checked.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
+      - name: owner
+        aliases:
+          - name: githubOrg
+        description: Name of the GitHub organization.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
+      - name: repository
+        aliases:
+          - name: githubRepo
+        description: Name of the GitHub repository.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
+      - name: requiredChecks
+        description: List of checks which have to be set to 'required' in the GitHub repository configuration.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: "[]string"
+      - name: requireEnforceAdmins
```

Comment:
I think for a `bool` we should have a default (`false`).

Reply:
the default of a bool is always false. Not sure if we have to mention this explicitly.
For `string` it is implicitly `""` without mentioning.

Reply:
Ok, maybe we need not to mention that in the `yaml` but add this to the docs generator.

Reply:
don't we have it there? If not, you are absolutely right!

Reply:
I'll check..